packages/nimble-components/src/button/specs/README.md

```diff
@@ -0,0 +1,45 @@
+# Nimble-Button New Visual Designs Implementation
+
+## Problem Statement
+
+The Visual Design spec for Nimble buttons now includes primary button states `primary-mono` and `primary` that aren't implemented in the nimble-button component yet.
+
+## Links To Relevant Work Items and Reference Material
+
+[Nimble Issue #663: Adopt nimble-button visual designs for states primary, quiet, and loud](https://github.com/ni/nimble/issues/663)
+
+[Figma Spec](https://www.figma.com/file/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?type=design&node-id=1295-77205)
+
+## Implementation / Design
+
+[FAST Button API](https://github.com/microsoft/fast/blob/de7f234ef871204fcac2b5df59433d919809341d/packages/web-components/fast-foundation/src/button/README.md)
+
+The appearance of the button will have states `default` (White), `primary-mono` (Black), and `primary` (Green), each with its own color scheme based on the UI theme.
```

Comment:
We try to avoid shorthands for property names -- [see docs here](https://github.com/ni/nimble/blob/main/packages/nimble-components/CONTRIBUTING.md#attribute-common-value-patterns). Assuming we're fine with the direction of `primary-mono` and `primary`, I think they really should be `primary-monochromatic` and `primary`.

It does make the value pretty long, but at least off the top of my head, I can't think of a better name suggestion.

Reply:
For reference, in the Figma spec the variants are called "p1" (monochromatic) and "p2" (colored). 

Assuming we agree that the color version should be the "first choice" primary state, I agree it makes sense to call it `primary` so that prominence is more obvious and so that existing clients migrate to it automatically. If that's the case, then some other name ideas for the grey one are:
- `primary-monochrome`
- `primary-neutral`
- `primary-quiet`
- `primary-2`

Whatever we pick, we should ask Brandon to reflect that decision in Figma.

Reply:
> Assuming we agree that the color version should be the "first choice" primary state, I agree it makes sense to call it primary so that prominence is more obvious and so that existing clients migrate to it automatically.

One question I have is should we expect only one style of primary button to be used on a page at a time / even a whole app at a time? Or should there be one primary color button on a page and maybe multiple primary monochrome buttons? i.e. individual sections could have monochrome buttons but a page should have a at most a single color button?

It seems weird to me that we would mix different styles of primary buttons across different button groups on a page or even a whole app.

Some other ideas (P1 vs P2):
- primary vs brand
- primary vs primary-brand
- primary-flat vs primary
- primary-subtle vs primary
- secondary vs primary (if pages should have have like multiple button groups with secondary buttons but like a single primary button on a page)

Reply:
And to complete the thought, if there should only be one style on a page at a time maybe it should be a theme provider configuration token instead so apps can't accidentally mix styles on a page (as easily)

Reply:
> > Assuming we agree that the color version should be the "first choice" primary state, I agree it makes sense to call it primary so that prominence is more obvious and so that existing clients migrate to it automatically.
> 
> One question I have is should we expect only one style of primary button to be used on a page at a time / even a whole app at a time? Or should there be one primary color button on a page and maybe multiple primary monochrome buttons? i.e. individual sections could have monochrome buttons but a page should have a at most a single color button?
> 
> It seems weird to me that we would mix different styles of primary buttons across different button groups on a page or even a whole app.

I'm a bit confused about this as well, I'll talk to Brandon about this today.